include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,370 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/copy.hpp>
+#include <eve/algo/equal.hpp>
+#include <eve/algo/fill.hpp>
+#include <eve/algo/views/convert.hpp>
+#include <eve/algo/container/detail/soa_storage2.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/product_type.hpp>
+#include <eve/traits.hpp>
+#include <eve/assert.hpp>
+
+#include "eve/algo/mismatch.hpp"
+#include <memory>
+
+namespace eve::algo
+{
+  struct no_init_t {};
+
+  //! Markup to trigger construction of container without initialisation of content
+  inline constexpr no_init_t no_init = {};
+
+  //================================================================================================
+  //! @addtogroup memory
+  //! @{
+  //!   @brief SIMD-aware container for product types
+  //!
+  //!   **Required header:** `#include <eve/memory/soa_vector.hpp>`
+  //!
+  //!   soa_vector is a container similar to `std::vector` but optimized to store
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template< eve::product_type Type
+          , eve::simd_allocator Allocator = eve::aligned_allocator<unsigned char, eve::fixed<64>>
+          >
+  struct soa_vector
+  {
+    public:
+
+    using storage_t = detail::soa_storage<Type,Allocator>;
```

Comment:
std calls them _type.
Not sure if this really public so w/e.

Reply:
It is not meant to be public, hence the `_t`

Reply:
If it's not meant to be public, don't make it?